### PR TITLE
Feature #2622 forecast climatology

### LINKED ISSUE #2622: Enhancement: Support separate climatology datasets for both the forecast and observation inputs
dtcenter/MET#2924 describes changes that will be made to MET to support forecast climo_mean and climo_stdev.

## Describe the Enhancement ##
Enhance the METplus wrappers to accompany the MET changes. From Slack, @JohnHalleyGotway noted:

> For METplus, I imagine we’d just want to…
> * Add support for FCST_CLIMO_MEAN and OBS_CLIMO_MEAN… and use them to set “climo_mean” inside the MET “fcst” dictionary.
> * Add support for FCST_CLIMO_STDEV and OBS_CLIMO_STDEV… and use them to set “climo_stdev” inside the MET “obs” dictionary.
> * If the FCST/OBS_CLIMO_MEAN/STDEV entries are not set, but CLIMO_MEAN/STDEV are, just set:
>   * FCST_CLIMO_MEAN = OBS_CLIMO_MEAN = CLIMO_MEAN
>   * FCST_CLIMO_STDEV = OBS_CLIMO_STDEV = CLIMO_STDEV

### Time Estimate ###
~1 day

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [X] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
6.0.0

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [X] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Review default **alert** labels
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Milestone and Projects ###
- [ ] Select **Milestone** as a **METplus-Wrappers-X.Y.Z** version, **Consider for Next Release**, or **Backlog of Development Ideas**
- [ ] For a **METplus-Wrappers-X.Y.Z** version, select the **METplus-Wrappers-X.Y.Z Development** project

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [ ] For any new datasets, an entry to the [METplus Verification Datasets Guide](https://metplus.readthedocs.io/en/latest/Verification_Datasets/index.html).
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next official version
Select: **METplus-Wrappers-X.Y.Z Development** project for development toward the next official release
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
**Note: #2695 should be resolved and truth data updated before this PR should be merged.**

Added support for setting fcst and obs specific climo_mean and climo_stdev dictionary values.

### Additional changes in this PR:
* Added documentation to describe VAR<n> config variables that are used to set climo field values
* Tighten up script run in GitHub Actions that occasionally fails to query DockerHub by retrying query and reporting a more useful error log message if the query still fails to alert developers to rerun the failed GHA job.
* Turn off use case groups that were left on from a previous PR
* Added .idea files generated by PyCharm
* Re-ordered assertions in conftest.py so that PyCharm correctly labels expected vs. actual values
* Minor fix to test_use_cases.py script to appease SonarQube complaint
* Removed unnecessary lines from use case Python scripts to satisfy SonarQube

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Added unit tests to ensure new variables work as expected

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review code changes and ensure all automated tests pass

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [X] Do these changes introduce new SonarQube findings? **[No]**</br>
If **yes**, please describe:

- [X] Please complete this pull request review by **9/25/2024**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [X] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [X] For any new datasets, an entry to the [METplus Verification Datasets Guide](https://metplus.readthedocs.io/en/latest/Verification_Datasets/index.html).
- [X] Review the source issue metadata (required labels, projects, and milestone).
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **METplus-Wrappers-X.Y.Z Development** project for official releases
- [X] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Review by JohnHalleyGotway — COMMENTED** [review]
@georgemccabe thanks for all of these changes. I do note that all of the GHA tests for this PR pass. My only question is about `use_case_groups.json`, where the `s2s_mid_lat` 0 - 3 jobs are now disabled. What's the reason for that change?

Here's a list of the change I see in this PR:
- Addition of the `.idea` files. No idea about the need for this but defer to you and note that this was intentional.
- Addition of docs about Ensemble-Stat fcst/obs climo settings.
- Addition of many new pytests.
- Updates to wrappers for ensemble_stat, gen_ens_prod, grid_stat, point_stat, series_analysis
- Updates to corresponding wrapped MET config files
- Updates to corresponding METplus conf files

**Comment by georgemccabe** [comment]
@JohnHalleyGotway, those use case groups were turned on for testing at some point in a feature branch and were accidentally left turned on, so they were running for every push to feature branches. They should be turned off before merging into develop. I thought about adding a GHA workflow to run on develop to use sed to turn them all off in develop to prevent this from happening.

**Review by JohnHalleyGotway — APPROVED** [review]
@georgemccabe thanks for confirming that change was intended. I approve of these changes.

**Pull request merged**
